### PR TITLE
Gofumpt all the things

### PR DESCRIPTION
**Pull request by ndeloof**
same as https://github.com/docker/compose/pull/12365 + make target

### CODE REVIEW

**Review by glours — APPROVED** [review]
LGTM
Open question, should not we add a check of the formatting during the validation step in the CI? 🤔

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ jobs:
           - validate-go-mod
           - validate-headers
           - validate-docs
+          - validate-fmt
```
Comment (thaJeztah) [inline .github/workflows/ci.yml:53]: FWIW; there's a golangci-lint linter for `gofumpt`; https://golangci-lint.run/usage/linters/#gofumpt

**Comment by thaJeztah** [comment]
Thanks! I started to add the same logic as we had in docker/cli in the Makefile, which does a "optimistic" try to see if `gofumpt` is installed, but then wasn't sure if that's what you wanted, or if you preferred a Dockerfile-based / bake target instead https://github.com/docker/cli/blob/8454e012cadeca7719f170012c5331655ed5e5ec/Makefile#L52-L58

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ndeloof) [reply .github/workflows/ci.yml]: nice, makes things even simpler

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**